config.go

```diff
@@ -703,27 +693,28 @@ func loadConfig() (*config, error) {
 		}
 	}
 
-	// At this point, we'll save the base data directory in order to ensure
-	// we don't store the macaroon database within any of the chain
-	// namespaced directories.
-	macaroonDatabaseDir = cfg.DataDir
+	// Compute network directory to store network specific
+	// files such as macaroon database, macaroons etc.
+	networkDir = filepath.Join(cfg.DataDir,
+		defaultChainSubDirname,
+		registeredChains.PrimaryChain().String(),
+		normalizeNetwork(activeNetParams.Name))
 
-	// If a custom macaroon directory wasn't specified and the data
-	// directory has changed from the default path, then we'll also update
-	// the path for the macaroons to be generated.
-	if cfg.DataDir != defaultDataDir && cfg.AdminMacPath == defaultAdminMacPath {
+	// If a custom macaroon path wasn't specified we'll
+	// store the macaroons within the network level.
+	if cfg.AdminMacPath == "." {
```

Comment:
`.` in Unix indicates the current directory - perhaps we should make `cleanAndExpandPath` return an empty string when given one?